server.js

```diff
@@ -1,33 +1,97 @@
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const path = require('path')
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/', (request, response) => {
+  response.sendFile(path.join(__dirname+'/index.html'))
+})
+
+app.get('/shows', (request, response) => {
+  const { cast, year, country } = request.query //?year=2019&country=South Korea
+  let movieList = netflixData;
+
+  movieList = filterMoviesOnYear(year, movieList);
+  movieList = filterMoviesByCountry(country, movieList);
+  movieList = filterMoviesByActor(cast, movieList);
+    
+  response.json(mappedNetflix(movieList))
+  
+})
+
+app.get('/shows/:id', (request, response) => {
+  const { id } = request.params
+  const movie = netflixData.find((item) => item.show_id === +id)
+
+  if (movie === undefined) {
+    response.status(404).json(`This show (${id}) does not exist!`)
+  }
+
+  response.json(movie)
+})
+
+app.get('/shows/:id/cast', (request, response) => {
+  const { id } = request.params
+  const movieId = netflixData.find((item) => item.show_id === +id)
+
+  if (movieId === undefined) {
+    response.status(404).json(`This show (${id}) does not exist!`)
+  }
+
+  response.json(movieId.cast.split(", "))
+})
+
+app.get('/categories', (request, response) => {
+  let category = [];
+
+  netflixData.forEach((item) => {
+    if (!category.includes(item.type)) {
+      category.push(item.type)
+    }
+    else return
+  })
+ 
+  response.json(category)
+})
+
+app.get('/categories/:category', (request, response) => {
+  const { cast, year, country, category } = request.query 
+
+  let filteredCategory = netflixData;
+
+  filteredCategory = filteredCategory.filter((item) => item.type.toLowerCase() === category.toLowerCase())
+
+  if (filteredCategory.length === 0) {
+    response.status(404).json(`This category (${category}) does not exist!`)
+    return
+  }
+
+  filteredCategory = filterMoviesOnYear(year, filteredCategory);
+  filteredCategory = filterMoviesByCountry(country, filteredCategory);
+  filteredCategory = filterMoviesByActor(cast, filteredCategory);
+
+  response.json(mappedNetflix(filteredCategory))
+
+})
+
+app.use(function (req, res, next) {
+  res.status(404).send(`Sorry can't find ${req.path}!`)
 })
 
-// Start the server
 app.listen(port, () => {
   console.log(`Server running on http://localhost:${port}`)
 })
+
+
+
+
+
+
```

Comment:
Look to your vscode settings and maybe you will find trimming function which will cut the tail of the blank lines on the end of your file ;)